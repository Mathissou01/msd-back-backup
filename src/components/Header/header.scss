@use "src/styles/01-settings/settings.colors" as colors;
@import "src/styles/02-tools/tools.z-index";

.c-Header {
  &__TopBar {
    position: fixed;
    top: 0;
    z-index: z(site-header);
    display: flex;
    flex-direction: row;
    width: 100vw;
    height: 60px;
    padding: 0 8px 0 20px;
    color: colors.$expert-blue;
    background: colors.$white;
    box-shadow: 0 0 1px rgb(17 95 251 / 5%), 0 1px 3px rgb(17 95 251 / 10%);
  }

  &__LeftBar {
    position: fixed;
    top: 60px;
    z-index: z(site-sidebar);
    display: flex;
    flex-direction: column;
    width: 207px;
    height: 100%;
    padding: 0;
    color: colors.$expert-blue;
    background-color: colors.$white;
    box-shadow: 0 0 1px rgb(17 95 251 / 5%), 0 1px 3px rgb(17 95 251 / 10%);
  }

  &__Nav {
    display: flex;
  }

  &__NavList {
    display: flex;
    margin: 0;
    padding: 0;
    list-style: none;

    &_top {
      gap: 50px;
      align-items: center;
    }

    &_side {
      flex-direction: column;
      align-items: flex-start;
    }
  }

  &__NavGroup {
    display: flex;
    flex-direction: column;
    width: 100%;
  }

  &__NavItem {
    display: flex;
    flex-direction: row;
    gap: 8px;
    align-items: center;
    height: 50px;
    padding: 0 6px 0 12px;
  }
}
