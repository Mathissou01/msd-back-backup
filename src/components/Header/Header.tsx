import Head from "next/head";
import Link from "next/link";
import "./header.scss";

export default function Header() {
  return (
    <>
      <Head>
        <title>MSD-BACK</title>
        <meta name="description" content="wip" />
        <link rel="icon" href="/images/favicon.ico" />
      </Head>

      <nav className={"c-Header__TopBar"} data-testid={"top-bar"}>
        <Link href={"/"}>
          <a style={{ fontWeight: "bold" }}>MSD BACK</a>
        </Link>
        <Link href={"/page-add"}>
          <a>Nouvelle Page</a>
        </Link>
        <Link href={"/gis"}>
          <a>GIS</a>
        </Link>
        <Link href={"/date-add"}>
          <a>Nouvelles Dates</a>
        </Link>
      </nav>
      <aside className={"c-Header__LeftBar"} data-testid={"left-bar"}>
        <Link href={`/agenda`}>
          <a style={{ textTransform: "uppercase" }}>Agenda</a>
        </Link>
        <div>MENU</div>
        <div>MENU</div>
        <div>MENU</div>
        <div>MENU</div>
        <div>MENU</div>
      </aside>
    </>
  );
}
