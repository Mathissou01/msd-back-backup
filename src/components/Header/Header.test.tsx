import { render, screen } from "@testing-library/react";
import Header from "./Header";

it("renders", () => {
  const { container } = render(<Header />);
  expect(container).toMatchSnapshot();
});

it("has navigation elements", () => {
  render(<Header />);
  const topBar = screen.getByTestId("top-bar");
  expect(topBar).toBeInTheDocument();
  const leftBar = screen.getByTestId("left-bar");
  expect(leftBar).toBeInTheDocument();
});
