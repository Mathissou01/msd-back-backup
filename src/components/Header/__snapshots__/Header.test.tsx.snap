// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <nav
    class="c-Header__TopBar"
    data-testid="top-bar"
  >
    <a
      href="/"
      style="font-weight: bold;"
    >
      MSD BACK
    </a>
    <a
      href="/page-add"
    >
      Nouvelle Page
    </a>
    <a
      href="/gis"
    >
      GIS
    </a>
    <a
      href="/date-add"
    >
      Nouvelles Dates
    </a>
  </nav>
  <aside
    class="c-Header__LeftBar"
    data-testid="left-bar"
  >
    <a
      href="/agenda"
      style="text-transform: uppercase;"
    >
      Agenda
    </a>
    <div>
      MENU
    </div>
    <div>
      MENU
    </div>
    <div>
      MENU
    </div>
    <div>
      MENU
    </div>
    <div>
      MENU
    </div>
  </aside>
</div>
`;
